scripts/check_urls.py

```diff
@@ -1,46 +1,51 @@
 #!/usr/bin/python3
 
-# Checks constructed download URLs and all other URLs it can find
-# Usage: ./check_urls.py plugins.yaml
-
+# Checks if `url` and `iconUrl` (if it exists) are valid
+# Usage: ./check_urls.py manifests/
+# Usage: ./check_urls.py myPlugin.yaml
+import os
 import yaml
 import sys
 from urllib.parse import urlparse
 from urllib.error import HTTPError
 from urllib.request import urlopen, Request
 
 
-def test(url):
-    print("Testing " + url)
+def check_url(url):
+    print("  Testing " + url)
     try:
         conn = urlopen(Request(url, method="HEAD"))
         return True
     except HTTPError as e:
-        print("%s: %s" % (e.code, e.reason))
+        print("  %s: %s" % (e.code, e.reason))
```

Comment:
I don't think i understand. Are you referring to the indentation, or the error formatting?

Reply:
There are two prints. Let's just have one print that has the information from both.

Reply:
To reduce the output, or the code size? Cause i can't think of a way to do this without more code.

Reply:
I suggested it so the success or failed state can be determined from one line instead of "maybe two"

An alternative would be for this method to return None on success, and the error message on error. Then the caller can log it, if they care to.

Reply:
How's this?

```py
def check_url(url):
    try:
        conn = urlopen(Request(url, method="HEAD"))
        status = "OK %s" % conn.status
        ok = False
    except HTTPError as e:
        status = "ERROR %s" % e.code
        ok = True
    
    print("  %s\t%s" % (status, url))
    return ok
```

```
Checking manifests/Ciu Kreo.yaml
  OK 200        https://github.com/RestingImmortal/Ciu-Kreo/archive/c8deb9aee5c7e0b96657d21e76b81602acbcb8a1.zip
  OK 200        https://github.com/RestingImmortal/Ciu-Kreo/raw/c8deb9aee5c7e0b96657d21e76b81602acbcb8a1/icon.png
Checking manifests/All Content Plugin.yaml
  ERROR 404     https://github.com/endless-sky/all-content-plugin/archive/refs/tasfdags/v0.9.13.zip
  OK 200        https://github.com/endless-sky/all-content-plugin/raw/v0.9.13/icon.png

```

Reply:
Sure (if your `ok` are meant to be like that). My concern was that grepping for a particular plugin wouldn't always get you the full context.